### ISSUE TITLE
Invalid Date String error

### ISSUE THREAD

**Issue by gjb83** [rsolr/rsolr#56]
When I add a document with a date to my Solr index, the date isn't automatically converted to the required ISO 8601 format:

```ruby
doc = {
  :id=>1,
  :my_string=>'foo',
  :my_int=>42,
  :my_bool=>true,
  :my_date=>Date.new(2013, 1, 1)
}
solr.add doc
```

The above example attempts to set the value of the `my_date` field to `2013-01-01`, which results in an error:

```javascript
{'responseHeader'=>{'status'=>400,'QTime'=>1},'error'=>{'msg'=>'Invalid Date String:\'2013-01-01\'','code'=>400}}
```

However, if I manually convert the date, it works fine:

```ruby
doc = {
  :id=>1,
  :my_string=>'foo',
  :my_int=>42,
  :my_bool=>true,
  :my_date=>Date.new(2013, 1, 1).to_time.utc.iso8601
}
solr.add doc
```

Since Solr only uses ISO 8601 format, is it possible to have RSolr convert Ruby `Date` and `Time` objects to the required format automatically?

I would be happy to work on a patch if you think this is a good idea.

**Comment by dchandekstark** [comment]
+1.  Autoconversion would alleviate downstream redundancy.

**Comment by cbeer** [comment]
Fixed by #125
